### PR TITLE
Add new structures and interfaces needed by sovereign chain

### PR DESCRIPTION
**Pull request by SebastianMarian**
* Added new structures and interfaces needed by sovereign chain

### CODE REVIEW

**Review by sasurobert — DISMISSED** [review]

File: data/block/shardHeaderExtended.go [hunk 1]
```diff
@@ -0,0 +1,894 @@
...
+		return data.ErrNilPointerReceiver
+	}
+	if she.Header == nil {
+		return data.ErrNilHeader
+	}
+
+	return she.Header.SetPrevRandSeed(pvRandSeed)
+}
+
+// SetRandSeed sets previous random seed
```
Comment (iulianpascalau) [inline data/block/shardHeaderExtended.go:342]: sets the random seed*

File: data/block/shardHeaderExtended.go [hunk 2]
```diff
@@ -0,0 +1,894 @@
...
+		ScheduledDeveloperFees:   devFees,
+		ScheduledGasProvided:     she.Header.GetScheduledGasProvided(),
+		ScheduledGasPenalized:    she.Header.GetScheduledGasPenalized(),
+		ScheduledGasRefunded:     she.Header.GetScheduledGasRefunded(),
+	}
+
+	return additionalVersionData
+}
+
+// GetValidatorStatsRootHash returns the root hash for the validator statistics trie
```
Comment (iulianpascalau) [inline data/block/shardHeaderExtended.go:824]: returns an empty byte slice*

File: data/block/sovereignChainHeader.go [hunk 3]
```diff
@@ -0,0 +1,530 @@
...
+func (sch *SovereignChainHeader) HasScheduledMiniBlocks() bool {
+	return false
+}
+
+// SetScheduledRootHash returns nil
+func (sch *SovereignChainHeader) SetScheduledRootHash(_ []byte) error {
+	return nil
+}
+
+// SetAdditionalData will not do anything
```
Comment (iulianpascalau) [inline data/block/sovereignChainHeader.go:27]: ... and return nil error*

File: data/block/sovereignChainHeader.go [hunk 4]
```diff
@@ -0,0 +1,530 @@
...
+	}
+
+	internalHeaderCopy := *sch.Header
+	headerCopy := *sch
+	headerCopy.Header = &internalHeaderCopy
+
+	return &headerCopy
+}
+
+// GetShardID returns the header shardID
```
Comment (iulianpascalau) [inline data/block/sovereignChainHeader.go:45]: returns 0 as the shardID*

File: data/block/sovereignChainHeader.go [hunk 5]
```diff
@@ -0,0 +1,530 @@
...
+// SetRound sets header round
+func (sch *SovereignChainHeader) SetRound(r uint64) error {
+	if sch == nil {
+		return data.ErrNilPointerReceiver
+	}
+
+	return sch.Header.SetRound(r)
+}
+
+// SetRootHash sets root hash
```
Comment (iulianpascalau) [inline data/block/sovereignChainHeader.go:256]: sets **the** root hash*

File: data/block/sovereignChainHeader.go [hunk 6]
```diff
@@ -0,0 +1,530 @@
...
+func (sch *SovereignChainHeader) SetRootHash(rHash []byte) error {
+	if sch == nil {
+		return data.ErrNilPointerReceiver
+	}
+
+	return sch.Header.SetRootHash(rHash)
+}
+
+// SetPrevHash sets prev hash
```
Comment (iulianpascalau) [inline data/block/sovereignChainHeader.go:265]: sets **the previous** hash

File: data/block/shardHeaderExtended.go [hunk 7]
```diff
@@ -0,0 +1,894 @@
...
+		return data.ErrNilPointerReceiver
+	}
+	if she.Header == nil {
+		return data.ErrNilHeader
+	}
+
+	return she.Header.SetRootHash(rHash)
+}
+
+// SetValidatorStatsRootHash sets the root hash for the validator statistics trie
```
[the remark this answers is not in the document]
Comment (SebastianMarian) [reply data/block/shardHeaderExtended.go:313]: Done

File: data/block/shardHeaderExtended.go (hunk 1 above)
Comment (SebastianMarian) [reply data/block/shardHeaderExtended.go:342]: Done

File: data/block/shardHeaderExtended.go (hunk 2 above)
Comment (SebastianMarian) [reply data/block/shardHeaderExtended.go:824]: Done

File: data/block/sovereignChainHeader.go (hunk 3 above)
Comment (SebastianMarian) [reply data/block/sovereignChainHeader.go:27]: Done

File: data/block/sovereignChainHeader.go (hunk 4 above)
Comment (SebastianMarian) [reply data/block/sovereignChainHeader.go:45]: Done

**Review by iulianpascalau — APPROVED** [review]

**Pull request merged**